testing/e2e/suites/operatorConfig.go

```diff
@@ -0,0 +1,96 @@
+package suites
+
+import (
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+)
+
+type cfgBuilder struct {
+	msi      string
+	tenantId string
+	location string
+}
+
+func builderFromInfra(infra infra.Provisioned) cfgBuilder {
+	return cfgBuilder{
+		msi:      infra.Cluster.GetClientId(),
+		tenantId: infra.TenantId,
+		location: infra.Cluster.GetLocation(),
+	}
+}
+
+type cfgBuilderWithOsm struct {
```

Comment:
out of curiosity - why implement this step-wise approach with new fields? why not just have one config struct with all the fields?

Reply:
one config struct with all fields does not work well with multiple dimensions.

The point of this builder is to easily support multiple dimensions which most tells will feature. This builder also doesn't stop someone from using a single config struct.



Example:

Say I'm defining that it works with the configs

- Feature A on, Feature B off
- Feature A off, Feature B off
- Feature A on, Feature B on
- Feature A off, Feature B on

That takes 4 definitions and if we add a new "feature" that will take 8 definitions

Better to use the builder pattern where we can say

Builder.WithFeatureA(on, off).WithFeatureB(on, off) which outputs the same thing